hail/src/main/scala/is/hail/expr/ir/agg/GroupedAggregator.scala

```diff
@@ -0,0 +1,259 @@
+package is.hail.expr.ir.agg
+
+import is.hail.annotations.{CodeOrdering, Region, RegionUtils, StagedRegionValueBuilder}
+import is.hail.asm4s._
+import is.hail.expr.ir.{EmitFunctionBuilder, EmitRegion, EmitTriplet, defaultValue, typeToTypeInfo}
+import is.hail.expr.types.physical._
+import is.hail.io._
+import is.hail.utils._
+
+class GroupedBTreeKey(kt: PType, fb: EmitFunctionBuilder[_], region: Code[Region], container: StateContainer) extends BTreeKey {
+  private val keyInline = kt.isPrimitive
+  val storageType: PStruct = PStruct(required = true,
+    "kt" -> (if (keyInline) kt else PInt64(kt.required)),
```

Comment:
I think it was originally to cap the size that a BTree key could be so that we wouldn't have to split nodes with an arbitrarily large amount of data to copy, but in practice I think any struct you'd want to use in this case is small enough to not be a concern. I'll remove this part of the logic.

Reply:
Ah, I see. Yeah, I suspect in practice the cost of the splits will be more than made up for by not having as many pointer dereferences when comparing keys.